Adapting to `ethersjs`

aaitor opened the pull request:
- adding command
- Adapting to `ethersjs`

## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

aaitor committed:
adding command

aaitor committed:
Adapting to `ethersjs`

aaitor committed:
linting and adding network governance commands

mrsmkl commented on src/commands/nfts/deployNft.ts:
```diff
@@ -39,37 +53,45 @@ export const deployNft = async (
+    const transactionResponse: TransactionResponse = await contract[
+      methodSignature
+    ](...args)
+    // ](...args, sendOptions)
```
clean up?

mrsmkl commented on src/utils/utils.ts:
```diff
@@ -55,14 +55,14 @@ export const loginMarketplaceApi = async (
 ): Contract => {
   const web3 = Web3Provider.getWeb3(config)
-  web3.setProvider(config.web3Provider)
+  // web3.setProvider(config.web3Provider)
```
cleanup?

mrsmkl commented on src/utils/utils.ts:
```diff
@@ -55,14 +55,14 @@ export const loginMarketplaceApi = async (
-  // @ts-ignore
-  return new web3.eth.Contract(abi, address)
+  return new ethers.Contract(address, abi, web3)
+  //return new web3.eth.Contract(abi, address)
```
cleanup?

mrsmkl reviewed (approved):
Looking good!

aaitor committed:
clean up

aaitor committed:
usage of accounts refactor

r-marques commented on README.md:
```diff
@@ -114,34 +114,51 @@ The CLI has pre-configured some Nevermined environments. You can check what is p
-   Metadata API: http://nevermined-metadata:5000
-   Faucet: http://localhost:3001
+
+ testnet:
```
I would probably name this `mumbai`

r-marques commented on README.md:
```diff
@@ -114,34 +114,51 @@ The CLI has pre-configured some Nevermined environments. You can check what is p
 A part of these networks you can connect to any other network using the environment variables described above.
 
-When you want to connect and interact with a different network of the default (`spree`), make sure you have exported the `NODE_URL` environment variable connected to the right environment, and pass the `-n NETWORK_NAME` parameter to your commands.
+When you want to connect and interact with a different network of the default (`testnet`), make sure you have exported the `NODE_URL` environment variable connected to the right environment, and pass the `-n NETWORK_NAME` parameter to your commands.
```
here too

r-marques commented on resources/commands.json:
```diff
@@ -9,7 +9,7 @@
 			"name": "network",
 			"alias": "n",
-			"default": "spree",
+			"default": "testnet",
```
here too

r-marques commented on resources/networks.json:
```diff
@@ -35,7 +35,7 @@
 		"networkId": "80001",
 		"networkName": "mumbai",
-		"contractsVersion": "2.0.0",
+		"contractsVersion": "2.0.3",
```
should this now be 2.0.4?

r-marques commented on src/cli.ts:
```diff
@@ -91,9 +91,9 @@ const cmdHandler = async (
-  const userAccount = findAccountOrFirst(accounts, argv.account)
-
+  // const accounts = await nvm.accounts.list()
+  // const userAccount = findAccountOrFirst(accounts, argv.account)
```
cleanup

r-marques commented on src/commands/accounts/list.ts:
```diff
@@ -50,16 +48,15 @@ export const accountsList = async (
-      const ethBalance = utils.fromWei(
-        (await a.getEtherBalance()).toString(),
-        'ether'
+      const ethBalance = ethers.utils.parseEther(
```
you can just call `BigNumber.parseEther`. trying to remove the need to call ethers directly when working with the sdk

r-marques commented on src/commands/assets/getAsset.ts:
```diff
@@ -3,7 +3,7 @@ import { StatusCodes } from '../../utils'
 import readline from 'readline'
 import { Logger } from 'log4js'
-import { makeKeyTransfer } from '@nevermined-io/nevermined-sdk-js/dist/node/utils'
+// import { makeKeyTransfer } from '@nevermined-io/nevermined-sdk-js/dist/node/utils'
```
cleanup

r-marques commented on src/commands/assets/getAsset.ts:
```diff
@@ -21,16 +21,17 @@ export const getAsset = async (
   let agreementId
 
-  if (password) {
-    const key = await keyTransfer.secretToPublic(keyTransfer.makeKey(password))
-    account.babyX = key.x
-    account.babyY = key.y
-    account.babySecret = password
-  }
+  // if (password) {
+  //   const key = await keyTransfer.secretToPublic(keyTransfer.makeKey(password))
+  //   account.babyX = key.x
+  //   account.babyY = key.y
+  //   account.babySecret = password
+  // }
```
maybe create a new issue to address this?

r-marques commented on src/commands/network/networkGetConfig.ts:
```diff
@@ -0,0 +1,69 @@
+import { ConfigEntry } from '../../models/ConfigDefinition'
+import { ethers } from 'ethers'
+
+export const networkGetConfig = async (
```
maybe rename this to `neverminedContractConfig` or `governanceConfig`. Right now we have different configs: the nevermined instance config which is also related to the network, and the nevermined config contract. It's important that the names don't raise confusion

r-marques commented on src/commands/nfts/burnNft.ts:
```diff
@@ -50,13 +50,14 @@ export const burnNft = async (
     )
 
-    const nft = loadNftContract(config, nftAddress)
+    const nft = loadNftContract(config, nftAddress).connect(config.signer)
```
there is no need for a custom function for this. you can use `nvm.contracts.loadNft721(address)`. The sdk also has the erc-721 abi so there is no need to have it on the cli

r-marques commented on src/commands/nfts/mintNft.ts:
```diff
@@ -55,13 +55,14 @@ export const mintNft = async (
     )
 
-    const nft = loadNftContract(config, nftAddress)
+    const nft = loadNftContract(config, nftAddress).connect(config.signer)
```
`nvm.contracs.loadNft721`

r-marques commented on src/utils/CustomToken.ts:
```diff
@@ -1,28 +1,33 @@
-    token.contract = new token.web3.eth.Contract(abi, address)
-
+    // token.contract = new token.web3.eth.Contract(abi, address)
+    // console.log(token.contract.)
```
cleanup

r-marques commented on src/utils/CustomToken.ts:
```diff
@@ -1,28 +1,33 @@
       throw new Error(`No code deployed at address ${address}, sorry.`)
     }
 
+    token.contract = loadERC20Contract(address, signer)
```
here we can also use `nvm.contracts.loadErc20(address)`

r-marques commented on src/utils/utils.ts:
```diff
@@ -34,7 +35,7 @@ export const loadNevermined = async (
     verbose: verbose ? verbose : config.nvm.verbose
   })
-
+  // console.log(`CONFIG NVM: ${JSON.stringify(config.nvm.web3Provider.)}`)
```
cleanup

r-marques commented on src/utils/utils.ts:
```diff
@@ -73,6 +71,26 @@ export const loadNftContract = (
+  signer: ethers.Signer
+): Contract => {
+  // const abi = `${ARTIFACTS_PATH}/NeverminedConfig.${config.networkName?.toLowerCase()}.json`
+  // const nvmConfigAbi = JSON.parse(fs.readFileSync(abiNvmConfig).toString())
```
cleanup

r-marques commented on src/utils/utils.ts:
```diff
@@ -250,11 +309,16 @@ export const printNativeTokenBanner = async () => {
+    // token.name(),
+    // token.symbol(),
+    // token.decimals(),
+    // token.totalSupply()
```
cleanup

r-marques reviewed (approved):
lgtm

r-marques committed:
Fix tests:

- infura ipfs now requires authentication headers
- added helper function print the error in case on commands fail
- improve fetch ipfs method to handle error conditions
- re-enabled polygon tests
- don't run tests in parallel to prevent failures because of the nounce

r-marques committed:
fix CI workflow

r-marques committed:
improved ipfs to work with and without auth

r-marques committed:
don't login for commands that don't require account

r-marques committed:
Fix CI for polygon tests

r-marques committed:
replaced loadNftContract with nvm loadNft721

r-marques committed:
use sdks customToken

- updated to latest version of the sdk

r-marques committed:
Fixed typo on CI env

- cleanup

r-marques committed:
Fix test issue where polygon-localnet accounts were not being funded
